### DISCUSSION TITLE
进入、退出插入模式、undo需要7秒...

### DISCUSSION THREAD

**Discussion by aztack** [nshen/learn-neovim-lua#85] [General]
进入、退出插入模式、undo需要7+秒...
通过排查后发现去掉autocmd.lua后恢复正常
MacOS Monterey 12.5

**Comment by nshen** [comment]
不会吧，再排查是哪个 autocmd 导致的呢？

**Comment by aztack** [comment]
另外，去掉autocmd，保存文件也需要等待5秒..
二分法排除了一下，注释掉下面git enabled 就好了。

```lua
git = {
  -- enabled = false, 
}
```
但是，注释掉之后左侧文件树按o首次打开文件夹时由有3秒延时...打开文件没有延时
尝试了一下把`enabled = true`，发现保存文件无延时，首次打开文件夹会延时。

发现官方有个issue：
https://github.com/kyazdani42/nvim-tree.lua/issues/549

我这边的原因是因为用的`fish`shell。在`basic.lua`里加这么一行：
```lua
vim.o.shell = '/bin/bash'
```

**Reply by aztack** [reply]
然而`st`打开的中断就是`bash`，`fish`下设置的变量都无法使用。需要手动切换
